uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/common/extensions/SdkExtensions.kt

```diff
@@ -173,4 +185,101 @@ fun Application.computePrimaryRoute(routePlan: RoutePlan, routeComputeCallback:
             override fun onPrimaryComputeFinished(router: Router, route: RouteInfo) = routeComputeCallback.invoke(route)
         })
     }
+}
+
+fun List<NaviSignInfo.SignElement>.concat(): String = StringBuilder().apply {
+    this@concat.forEach {
+        if (isNotEmpty()) append(", ")
+        append(it.text)
+    }
+}.toString()
+
+fun List<NaviSignInfo>.getNaviSignInfoOnRoute(): NaviSignInfo? {
+    this.filter { it.isOnRoute }.let { isOnRouteList ->
+        if (isOnRouteList.isEmpty()) {
+            return null
+        }
+
+        return isOnRouteList.firstOrNull { it.backgroundColor != 0 }?.let { it } ?: isOnRouteList[0]
+    }
+}
+
+fun NaviSignInfo.roadSigns(maxRoadSignsCount: Int = 3): List<RoadSignData> {
+    val roadSigns = mutableListOf<RoadSignData>()
+    val hasPic = signElements.find { it.elementType == SignElementType.Pictogram } != null
+    signElements
+        .asSequence()
+        .filter { it.elementType == SignElementType.RouteNumber }
+        .take(if (hasPic) maxRoadSignsCount - 1 else maxRoadSignsCount)
+        .toList()
+        .forEach { signElement ->
+            with(signElement.routeNumberFormat) {
+                if (insideNumber.isNotEmpty()) {
```

Comment:
tento if by som zakomponoval ako dalsi operator `sequence`
`filter(it.routeNumberFormat.insideNumber.isNotEmpty())`

Reply:
přemýšlím, zda ho nesloučit s tím prvním filtrem...

Reply:
tych filtrov v **sequence** mozes mat viac a nezmeni to performance, radsej daj 2, ako 1 s nejakou komplikovanou, zlozitou podmienkou. Pre collection by to mohlo mat zmysel, ale pre sequence radsje 2

Reply:
oki, tak dám